test/facility-guarantee/post-facility-guarantee.api-test.ts

```diff
@@ -0,0 +1,302 @@
+    expect(acbsRequest.isDone()).toBe(true);
+  });
+
+  it('rounds the maximumLiability to 2dp', async () => {
+    const requestBodyWithMaximumLiabilityToRound = [{ ...requestBodyToCreateFacilityGuarantee[0], maximumLiability: 1.234 }];
+    const acbsRequestBodyWithRoundedMaximumLiability = {
+      ...acbsRequestBodyToCreateFacilityGuarantee,
+      GuaranteedLimit: 1.23,
+    };
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsRequestWithRoundedMaximumLiability = requestToCreateFacilityGuaranteeInAcbsWithBody(acbsRequestBodyWithRoundedMaximumLiability).reply(
+      201,
+      undefined,
+      {
+        location: `/Portfolio/${portfolioIdentifier}/Facility/${facilityIdentifier}/FacilityGuarantee?accountOwnerIdentifier=00000000&lenderTypeCode=${lenderTypeCode}&sectionIdentifier=${sectionIdentifier}&limitTypeCode=${limitTypeCode}&limitKey=${limitKey}&guarantorPartyIdentifier=${guarantorParty}`,
+      },
+    );
+
+    const { status, body } = await api.post(createFacilityGuaranteeUrl, requestBodyWithMaximumLiabilityToRound);
+
+    expect(status).toBe(201);
+    expect(body).toStrictEqual({
+      facilityIdentifier,
+    });
+    expect(acbsRequestWithRoundedMaximumLiability.isDone()).toBe(true);
+  });
+
+  it(`replaces the effectiveDate with today's date if the specified effectiveDate is in future`, async () => {
+    const requestBodyWithFutureEffectiveDate = [{ ...requestBodyToCreateFacilityGuarantee[0], effectiveDate: '3000-01-02' }];
+    const acbsRequestBodyWithTodayEffectiveDate = {
+      ...acbsRequestBodyToCreateFacilityGuarantee,
+      EffectiveDate: new Date().toISOString().split('T')[0] + 'T00:00:00Z',
+    };
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsRequestWithTodayEffectiveDate = requestToCreateFacilityGuaranteeInAcbsWithBody(acbsRequestBodyWithTodayEffectiveDate).reply(201, undefined, {
+      location: `/Portfolio/${portfolioIdentifier}/Facility/${facilityIdentifier}/FacilityGuarantee?accountOwnerIdentifier=00000000&lenderTypeCode=${lenderTypeCode}&sectionIdentifier=${sectionIdentifier}&limitTypeCode=${limitTypeCode}&limitKey=${limitKey}&guarantorPartyIdentifier=${guarantorParty}`,
+    });
+
+    const { status, body } = await api.post(createFacilityGuaranteeUrl, requestBodyWithFutureEffectiveDate);
+
+    expect(status).toBe(201);
+    expect(body).toStrictEqual({
+      facilityIdentifier,
+    });
+    expect(acbsRequestWithTodayEffectiveDate.isDone()).toBe(true);
+  });
+
+  withStringFieldValidationApiTests({
+    fieldName: 'facilityIdentifier',
+    length: 10,
+    required: true,
+    generateFieldValueOfLength: (length: number) => valueGenerator.facilityId(length - 4),
+    validRequestBody: requestBodyToCreateFacilityGuarantee,
+    makeRequest: (body) => api.post(createFacilityGuaranteeUrl, body),
+    givenAnyRequestBodyWouldSucceed: () => {
+      givenAuthenticationWithTheIdpSucceeds();
+      givenAnyRequestBodyToCreateFacilityGuaranteeInAcbsSucceeds();
+    },
+  });
```

Comment:
We have a `withFacilityIdentifierFieldValidationApiTests` function for this now

Reply:
Changed.